lib/iex/endpoints/historial_prices.rb

```diff
@@ -0,0 +1,28 @@
+module IEX
+  module Endpoints
+    module HistoricalPrices
+      def historical_prices(symbol, options = {})
+        if options[:range] == 'date'
+          raise ArgumentError if options[:date].nil?
+          raise ArgumentError if options[:chartByDay].nil?
+        end
+
+        options = options.dup
+        path = "stock/#{symbol}/chart"
+        path += "/#{options[:range]}" if options[:range]
```

Comment:
`if options.key?(:range)`, which is faster

Reply:
My bad, misunderstood your previous comment.